source/Halibut/Transport/SecureConnection.cs

```diff
@@ -32,15 +32,18 @@ public bool HasExpired()
         {
             return lastUsed < DateTimeOffset.UtcNow.Subtract(HalibutLimits.SafeTcpClientPooledConnectionTimeout);
         }
-
+        
         public void Dispose()
         {
             try
             {
                 protocol.StopAcceptingClientRequests();
                 try
                 {
+                    // TODO - ASYNC ME UP!
```

Comment:
For another PR

Reply:
Not sure what to do in the `Dispose` I guess we can use Dispose async with async usings. From memory the ValueTask can be a pita to work with.

Reply:
I haven't investigated, just flagged so we revisit and determine if there is anything we want to change 👍